larray/tests/test_array.py

```diff
@@ -706,108 +706,117 @@ def test_getitem_structured_key_with_groups():
 
 
 def test_getitem_single_larray_key_guess():
-    a = Axis(['a1', 'a2'], 'a')
-    b = Axis(['b1', 'b2', 'b3'], 'b')
-    c = Axis(['c1', 'c2', 'c3', 'c4'], 'c')
+    # TODO: we really need another way to get test axes, e.g. testaxes(2, 3, 4) or testaxes((2, 3, 4))
+    a, b, c = ndtest((2, 3, 4)).axes
+    arr = ndtest((a, b))
+    # >>> arr
+    # a\b b0 b1 b2
+    #  a0  0  1  2
+    #  a1  3  4  5
 
     # 1) key with extra axis
-    arr = ndtest([a, b])
-    # replace the values_axis by the extra axis
-    key = LArray(['a1', 'a2', 'a2', 'a1'], [c])
-    assert arr[key].axes == [c, b]
-
-    # 2) key with the values axis (the one being replaced)
-    arr = ndtest([a, b])
-    key = LArray(['b2', 'b1', 'b3'], [b])
+    key = LArray(['a0', 'a1', 'a1', 'a0'], c)
+    # replace the target axis by the extra axis
+    expected = from_string(r"""
+c\b  b0  b1  b2
+ c0   0   1   2
+ c1   3   4   5
+ c2   3   4   5
+ c3   0   1   2""")
+    assert_array_equal(arr[key], expected)
+
+    # 2) key with the target axis (the one being replaced)
+    key = LArray(['b1', 'b0', 'b2'], b)
     # axis stays the same but data should be flipped/shuffled
-    assert arr[key].axes == [a, b]
+    expected = from_string(r"""
+a\b  b0  b1  b2
+ a0   1   0   2
+ a1   4   3   5""")
+    assert_array_equal(arr[key], expected)
+
+    # 2bis) key with part of the target axis (the one being replaced)
+    key = LArray(['b2', 'b1'], 'b=b0,b1')
+    expected = from_string(r"""
+a\b  b0  b1
+ a0   2   1
+ a1   5   4""")
+    assert_array_equal(arr[key], expected)
+
+    # 3) key with another existing axis (not the target axis)
+    key = LArray(['a0', 'a1', 'a0'], b)
+    expected = from_string("""
+b  b0  b1  b2
+\t  0   4   2""")
+    assert_array_equal(arr[key], expected)
+
+    # TODO: this does not work yet but should be much easier to implement with "align" in make_np_broadcastable
+    # 3bis) key with *part* of another existing axis (not the target axis)
+    # key = LArray(['a1', 'a0'], 'b=b0,b1')
+    # expected = from_string("""
+    # b  b0  b1
+    # \t  3   1""")
+    # assert_array_equal(arr[key], expected)
+
+    # 4) key has both the target axis and another existing axis
+    # TODO: maybe we should make this work without requiring astype!
```

Comment:
why not passing `dtype=str` to `from_string`? It should work no?

Reply:
It's a good idea which I didn't think about... but sadly it does not work:
```python
>>> from_string(r"""
... a\b b0 b1 b2
...  a0 a0 a1 a0
...  a1 a1 a0 a1""", dtype=str).dtype
dtype('O')

>>> from_string(r"""
... a\b b0 b1 b2
...  a0 a0 a1 a0
...  a1 a1 a0 a1""").astype(str).dtype
dtype('<U2')
```
Note that using dtype=np.str_ or dtype='<U2' does not work either.
TypeError: the dtype <U2 is not supported for parsing.